### DISCUSSION TITLE
Sync issues with changing external IP from ISP + workaround solution

### DISCUSSION THREAD

**Discussion by Daniel-1276** [Chia-Network/chia-blockchain#5469] [General]
**Describe the bug**
In germany we get disconnected with our IPv4 addresses every 24h. After disconnect and reconnect we do get a new external IP. As soon as this happens at a certain time in the night, sync of chia full client gets lost.

**To Reproduce**
- reboot router
- change external IP

**Expected behavior**
Chia client should be able to reconnect by itself after external IP change

**Screenshots**
n/a

**Desktop (please complete the following information):**

- OS: Windows 10
- Port 8444 open from external

**Additional context**
Possible workaround below for FritzBox users developed. Windows Powershell-Script checks for external IP-change on FritzBox router via UPNP / SOAP request and will initiate a restart via "./chia.exe start farmer -r". Tested on FritzBox 7530, maybe this helps somebody until final fix.

```
function check-ip{
param (
	[String] $URL = "http://192.168.178.1:49000/igdupnp/control/WANIPConn1"
)

$soapWebRequest = [System.Net.WebRequest]::Create($URL) 
$soapWebRequest.Headers.Add("SOAPAction",'"urn:schemas-upnp-org:service:WANIPConnection:1#GetExternalIPAddress"')

$soapWebRequest.ContentType = 'text/xml;charset="utf-8"'
$soapWebRequest.Accept      = "text/xml" 
$soapWebRequest.Method      = "POST" 

$requestStream = $soapWebRequest.GetRequestStream() 
[string]$SOAPRequest='<?xml version="1.0" encoding="utf-8"?>
<s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/">
<s:Body>
  <u:GetExternalIPAddress xmlns:u=urn:schemas-upnp-org:service:WANIPConnection:1 />
</s:Body>
</s:Envelope>';

$SOAPRequestbytearray = ([system.Text.Encoding]::ASCII).GetBytes($SOAPRequest)
$requestStream.write($SOAPRequestbytearray,0,$SOAPRequestbytearray.count)
$requestStream.Close() 

$resp = $soapWebRequest.GetResponse() 
$responseStream = $resp.GetResponseStream() 
$soapReader = [System.IO.StreamReader]($responseStream) 
$ReturnXml = [Xml] $soapReader.ReadToEnd() 
$responseStream.Close()

$ip=$ReturnXml.Envelope.Body.GetExternalIPAddressResponse|foreach { $_.NewExternalIPAddress }
return $ip
}

# VARIABLES
$localappdata=$env:LOCALAPPDATA
$chiapath="chia-blockchain\app-1.1.5\resources\app.asar.unpacked\daemon"
$currentIP=check-ip
$lastIP=$currentIP

if (Test-Path -Path $localappdata\$chiapath\chia.exe){
write-host "Path to chia.exe checked and valid"
}else{
write-host "Path to chia.exe NOT valid, please check script"
pause
exit
}
write-host ""
write-host "Your current FritzBox external IP: $currentIP"
write-host "Start monitoring " -NoNewline

while($true){
	if(test-connection 8.8.8.8 -quiet){
		$currentIP = check-ip
	}
	if ($currentIP -ne $lastIP){
		cls
		write-host "IP change detected !"
		write-host "Old IP: $lastIP New IP: $currentIP"
		$lastIP = $currentIP

        	write-host "Restarting chia client now ..."
		start-process $localappdata\$chiapath\chia.exe -argumentlist "start farmer -r"

		write-host ""
		write-host "Start monitoring " -NoNewLine
	}
    Start-Sleep -Seconds 360
    write-host "." -NoNewline
}
```

**Comment by boombel666** [comment]
"+"
I live in Germany with the same problem.  everynight i got disconnecting with node because my ISP changing my ip. After this  i have to manual restart chia app. I am plotting and farming on the same pc so when im restarting app my plots in paralel are canceled. I am using Huawei B525s-23a. Maybe someone can tell me, is it possible to restart the app without loosing plotting queue?

**Comment by Marqitos** [comment]
"+"
I live in spain with the same problem. I wait to finish de plotting queue to restart de app, but is a bad solution.
Thanks for the help.

**Comment by Daniel-1276** [comment]
I would strongly recommend using powershell as well for plotting ! If you do, you have no issues restarting the GUI. On the downside you don't get fancy progress bars ;-)

Commandline I use in Powershell below. 
```
cd C:\Users\***\AppData\Local\chia-blockchain\app-1.1.5\resources\app.asar.unpacked\daemon
start-process ./chia.exe -argumentlist "plots create -k 32 -b 4608 -u 128 -r 2 -t D:\FASTSSD -2 E:\SLOWER_SSD -d F:\BIG_SATA_HDD -n 10"
Start-Sleep -Seconds 3600
start-process ./chia.exe -argumentlist "plots create -k 32 -b 4608 -u 128 -r 2 -t D:\FASTSSD -2 E:\SLOWER_SSD -d F:\BIG_SATA_HDD -n 10"
Start-Sleep -Seconds 3600
start-process ./chia.exe -argumentlist "plots create -k 32 -b 4608 -u 128 -r 2 -t D:\FASTSSD -2 E:\SLOWER_SSD -d F:\BIG_SATA_HDD -n 10"
Start-Sleep -Seconds 3600
start-process ./chia.exe -argumentlist "plots create -k 32 -b 4608 -u 128 -r 2 -t D:\FASTSSD -2 E:\SLOWER_SSD -d F:\BIG_SATA_HDD -n 10"
```
You can add as many parallel plotters as your SSD / CPU / RAM can take ;-) It is important to have some time in between, else all plots will write to slow SATA HDD at once.

**Comment by boombel666** [comment]
Thank You so much. i Have only two question.
Why are You using  "E:\SLOWER_SSD -d"  ? in gui there is only 2 directories for temp(ssd) and final plot(hdd) files.
is "Start-Sleep -Seconds 3600" delay between plots from GUI?

**Comment by Daniel-1276** [comment]
same here in the CLI. "-t" is the first temp directory "-2" is the second temporary location (like in the GUI) and "-d" defines final directory. You can leave out "-2" location if do not have separate "slow" SSD's. Check https://github.com/Chia-Network/chia-blockchain/wiki/CLI-Commands-Reference for all CLI options.

**Comment by joelrb** [comment]
The plotting not lost after restarting using your script?

**Comment by joelrb** [comment]
Add your comment to my request to just decouple plotting from the rest so we can restart Chia GUI anytime without losing plotting in progress.

https://github.com/Chia-Network/chia-blockchain/discussions/5467
